shakedown/cli/main.py

```diff
@@ -57,6 +68,42 @@ def cli(**args):
     if not args['no_banner']:
         echo(banner(), n=False)
 
+    if args['provisioner_url']:
+      echo('Provisioning DC/OS cluster...', d='step-maj')
+
+      echo('Checking provisioner URL...', d='step-min', n=False)
+      try:
+        requests.get(args['provisioner_url'])
+        echo(args['provisioner_url'])
+      except:
+        click.secho("error: cannot connect to '{}'.".format(args['provisioner_url']), fg='red', bold=True)
+        sys.exit(1)
+
+      echo('Checking requested cluster platform...', d='step-min', n=False)
+      echo(args['provisioner_platform'])
+
+      success = False
+      attempt = 1
+      while not success:
+        echo("Waiting for cluster...", d='step-min', n=False)
+
+        response = requests.post("{}/api/v1/vm/{}".format(args['provisioner_url'], args['provisioner_platform'])).json()
+        if ('ok' in response) and (response['ok']):
+          success = True
+          args['cluster_id'] = response[args['provisioner_platform']]['vm']
+          echo(args['cluster_id'])
+          args['dcos_url'] = 'http://' + requests.get("{}/api/v1/vm/{}".format(args['provisioner_url'], args['cluster_id'])).json()[args['cluster_id']]['hostname']
+        else:
+          if attempt < (args['provisioner_max_tries'] + 1):
```

Comment:
It looks like this logic will work as long as vmpooler either has or is able to provide a cluster within 60 + 120 + 180 = 360 seconds. Does it make sense to have a longer default wait time and a fixed number of times (with no exponential back off)?